lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -139,15 +148,18 @@ def queue_job(self, job_wrapper):
             k8s_job_prefix,
             self.__get_k8s_job_spec(ajs)
         )
-
         job = Job(self._pykube_api, k8s_job_obj)
-        job.create()
-        job_id = job.metadata['name']
+        try:
+            job.create()
+            job_id = job.metadata.get('name')
+            # define job attributes in the AsyncronousJobState for follow-up
+            ajs.job_id = job_id
+            # store runner information for tracking if Galaxy restarts
+            job_wrapper.set_external_id(job_id)
+        except Exception as e:
+            log.debug(str(e))
```

Comment:
If this is not supposed to happen if everything is working "normally" I would do this
```suggestion
            log.exception("Error occurred while creating kubernetes job object")
```
That'll include the traceback, and depending on the logging setup various additional things that might be helpful for debugging this further.